src/node.rs

```diff
@@ -349,7 +347,7 @@ impl Taffy {
 
     /// Indicates whether the layout of this node (and its children) need to be recomputed
     pub fn dirty(&self, node: Node) -> TaffyResult<bool> {
-        Ok(self.nodes[node].is_dirty)
+        Ok(self.nodes[node].size_cache.iter().all(|entry| entry.is_none()))
```

Comment:
It makes a lot more sense to just use the "source truth" directly instead of having an extra boolean to keep track of.